client/dashboard/app/dataviews/use-persistent-view.ts

```diff
@@ -81,6 +83,21 @@ export function usePersistentView( {
 		// eslint-disable-next-line react-hooks/exhaustive-deps
 	}, [ JSON.stringify( transientFilterFields ) ] );
 
+	useEffect( () => {
+		let transientQueryParams: Record< string, unknown > = {};
+		transientFilters.forEach( ( { field, value } ) => {
+			transientQueryParams[ field ] = value;
+		} );
+		transientQueryParams = mergeQueryParamsWithTransientProperties(
+			transientQueryParams,
+			transientProperties
+		);
+		setTransientQueryParamsAtPathname(
+			matches[ matches.length - 1 ].pathname.replace( /\/$/, '' ),
```

Comment:
Can matches be empty?

Reply:
I assumed not apparently it could be in some edge cases... so I'm guarding it now.